core/providers/azure.go

```diff
@@ -63,40 +64,57 @@ func (provider *AzureProvider) GetProviderKey() schemas.ModelProvider {
 // completeRequest sends a request to Azure's API and handles the response.
 // It constructs the API URL, sets up authentication, and processes the response.
 // Returns the response body, request latency, or an error if the request fails.
-func (provider *AzureProvider) completeRequest(ctx context.Context, requestBody interface{}, path string, key schemas.Key, model string) ([]byte, time.Duration, *schemas.BifrostError) {
+// Parameters:
+//   - method: HTTP method (GET, POST, etc.)
+//   - requestBody: request body (can be nil for GET requests)
+//   - path: API path (e.g., "chat/completions", "models")
+//   - key: Azure authentication key
+//   - model: model name (can be empty for non-deployment endpoints)
+//   - useDeployment: whether to use deployment-based URL construction
+func (provider *AzureProvider) completeRequest(ctx context.Context, method string, requestBody interface{}, path string, key schemas.Key, model string, useDeployment bool) ([]byte, time.Duration, *schemas.BifrostError) {
```

Comment:
we can follow the same convention of req.Type instead of a useDeployement boolean

Reply:
or we can simply check for "models" and not usedeployment,  an extra variable is not needed